src/main/java/at/ac/tuwien/kr/alpha/solver/NaiveSolver.java

```diff
@@ -277,8 +278,22 @@ private void doBacktrack() {
 		}
 	}
 
+	private static IntIterator fromIterator(Iterator<Integer> it) {
+		return new IntIterator() {
+			@Override
+			public boolean hasNext() {
+				return it.hasNext();
+			}
+
+			@Override
+			public int next() {
```

Comment:
Not sure if this method makes sense.. `it.next()` gives an `Integer` (already boxed), and here it's unboxed again, so an `IntIterator` that is obtained through this method probably does not reduce autoboxing. It may be that the compiler is smart enough to see that the boxing/unboxing could be skipped, but I suspect that is not the case (thanks to type erasure). Hard to verify wiithout reading bytecode though..

Reply:
Your analysis is correct, this method does not avoid `Integer` objects. The surrounding anonymous class just wraps an `Iterator<Integer>` inside an `IntIterator` for the sole reason of using it inside the `NaiveSolver`, which is a totally non-optimized (and simple) version of a solver anyway.